File: Unit2-01/Java/MrCoxallStack.java
/*
* This class defines a string stack
*
* @author  Evan Beaudoin
* @version 1.0
* @since   2024-03-08
*/

import java.util.ArrayList;
import java.util.List;

/**
 * This is the class.
*/
public class MrCoxallStack {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
    */
    private List<String> stackAsList;

    /**
     * Constructs a new instance of MrCoxallStack.
     * Initializes the stack with an empty ArrayList.
     *
    */
    public MrCoxallStack() {
        this.stackAsList = new ArrayList<>();
    }

    /**
     * Pushes the string to a list.
     *
     * @param string the string that is going to be pushed
    */
    public void push(String string) {
        this.stackAsList.add(string);
    }

    /**
     * Returns the stack value.
     *
     * @return the values of the list formatted
    */
    public String showStack() {
        final StringBuilder stackValues = new StringBuilder();

        for (String value : this.stackAsList) {
            stackValues.append(value).append(", ");
        }

        if (stackValues.length() > 0) {
            // Remove the trailing comma
            stackValues.deleteCharAt(stackValues.length() - 2);
        }

        return stackValues.toString();
    }
}

File: Unit2-01/Typescript/MrCoxallStack.ts
/**
* This is the MrCoxallStack class
*
* By:      Evan Beaudoin
* Version: 1.0
* Since:   2024-03-25
*/ 

export class MrCoxallStack {
  // Stack that contains all of the string
  private stackAsList

  // Creates the stack
  public constructor() {
    this.stackAsList = []
  }

  // Pushes the values to the stack
  public push(stringPushed) {
    this.stackAsList.push(stringPushed)
  }

  // Returns the stack
  public showStack() {
    let stackItems = ""
    
    for (let counter = 0; counter < this.stackAsList.length; counter++) {
      // If statments ensures that comma is not printed at the start
      if (counter === 0) {
        stackItems += this.stackAsList[counter]
      } else {
        stackItems += ", " + this.stackAsList[counter]
      }
    }

    return stackItems
  }
}
